akkeeper/src/main/scala/akkeeper/master/service/HeartbeatService.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2019 Iaroslav Zeigerman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package akkeeper.master.service
+
+import akka.actor.{Actor, ActorLogging, ActorRef, ActorRefFactory, Cancellable, Props}
+import akkeeper.api.{Heartbeat, TerminateMaster}
+import akkeeper.config._
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
+import HeartbeatService._
+
+private[akkeeper] final class HeartbeatService extends Actor with ActorLogging {
+
+  private implicit val dispatcher = context.dispatcher
+
+  private val heartbeatConfig: HeartbeatConfig = context.system.settings.config.master.heartbeat
+  private val timeoutTaskInterval: FiniteDuration = Duration.fromNanos(
+    (heartbeatConfig.interval * heartbeatConfig.timeoutFactor).toNanos)
```

Comment:
Why do you need these two parameters (interval and timeoutFactor), if they are always used together? How it would be different if you had single interval parameter set to 30*1.2=36?